utils/gitreader.go

```diff
@@ -44,37 +47,47 @@ func NewGitReader(projectName string, record DependencyRecord) *GitReader {
 	gitReader.Url = record.BaseRepo
 	gitReader.Version = record.Version
 	gitReader.ProjectPath = record.ProjectPath
+	gitReader.packageName = record.Packagename
 
 	return &gitReader
 
 }
 
 func (reader *GitReader) CloneDependency() error {
-	zipFileName := reader.Name + "." + reader.Version + ".zip"
+
+	zipFilePrefix := reader.Name + "." + reader.Version + ".zip."
 	zipFilePath := reader.Url + "/zipball" + "/" + reader.Version
 
-	os.MkdirAll(reader.ProjectPath, os.ModePerm)
-	output, err := os.Create(path.Join(reader.ProjectPath, zipFileName))
+	projectPath := reader.ProjectPath
+	os.MkdirAll(projectPath, os.ModePerm)
+
+	zipFile, err := ioutil.TempFile(projectPath, zipFilePrefix)
 	if err != nil {
-        return err
-    }
-	defer output.Close()
+		return err
+	}
+	zipFileName := zipFile.Name()
+	defer os.Remove(zipFileName)
```

Comment:
Good. "It is the caller's responsibility to remove the file when no longer needed."